QueryManager/src/main/java/eu/neclab/ngsildbroker/queryhandler/services/QueryService.java

```diff
@@ -948,8 +952,30 @@ private List<Uni<List<Object>>> getRemoteCalls(RowSet<Row> rows, String endpoint
 	}
 
 	public Uni<Map<String, Object>> retrieveEntity(Context context, String tenant, String entityId,
-			AttrsQueryTerm attrsQuery, LanguageQueryTerm lang, boolean localOnly) {
-		Uni<Map<String, Object>> local = queryDAO.getEntity(entityId, tenant, attrsQuery);
+												   String attrs, LanguageQueryTerm lang, boolean localOnly, String containedBy, String join,
+												   boolean idsOnly, int joinLevel) {
+		Map<String,Object> attrsMap = QueryParser.parseInput(attrs);
+		AttrsQueryTerm attrsQuery;
+		try {
+			attrsQuery = QueryParser.parseAttrs(attrs, context);
+		} catch (ResponseException e) {
+			throw new RuntimeException(e);
+		}
+		Uni<Map<String, Object>> local;
+		if(idsOnly){
+			//TODO
+			local = idsOnly(getEntityFlat(tenant, entityId, attrsMap, containedBy,  joinLevel, null, context));
+		}
+		else if (join==null) {
+			local = queryDAO.getEntity(entityId, tenant, attrsQuery);
+		}
+		else if (join.equals("flat")) {
```

Comment:
Please put in constants file like as "flat" and "inline"

Reply:
done